### ISSUE TITLE
Display cluster when there are more than 100(?) markers

### ISSUE THREAD

**Issue by vaurdan** [AHAAAAAAA/PokemonGo-Map#2199]
When you have a lot of scanning areas, the map starts to get slower if you zoom out because of the amount of pokemons, pokestops and gyms.

We could use something like Google Maps Clusterer to group clusters of markers when there are more than 100 markers being displayed: 
https://github.com/googlemaps/js-marker-clusterer

This would make the map much faster to use.

Ideas?

**Comment by Hoekynl** [comment]
Please add a toggle or option to disable this, its a nice feature but I wouldn't want it on

**Comment by DeirhX** [comment]
I agree with it being a nice and optional feature. For larger-scale search scenarios, thousands of objects now completely stall a map so we need a way to keep it operable. Smaller-scale usage however can do without.

**Comment by Plop** [comment]
Please someone do this. With the beehiver we will need it.

**Comment by vaurdan** [comment]
I'm trying to implement this. Once I have a working and tested solution I'm going to open a PR.
